src/main.rs

```diff
@@ -1,18 +1,161 @@
 use macroquad::prelude::*;
 
+const UPDATE_INTERVAL: f64 = 0.3;
+
+const SPACE_FOR_TEXT: i32 = 200;
+
 mod conways;
 
-#[macroquad::main("BasicShapes")]
+use conways::*;
+
+fn conf() -> Conf {
+    Conf {
+        window_title: "Conway's Game of Life".to_string(),
+        window_height: 600 + SPACE_FOR_TEXT,
+        window_width: 600,
+        fullscreen: false,
+        window_resizable: false,
+        ..Default::default()
+    }
+}
+
+#[macroquad::main(conf)]
 async fn main() {
+    let mut keys: [KeyCode; 10] = [KeyCode::Space; 10];
+    let mut offset = 0;
+
+    let mut grid = Grid::new(22);
+
+    // Glider 
+    grid.change_state((2, 1), State::Alive);
+    grid.change_state((2, 2), State::Alive);
+    grid.change_state((2, 3), State::Alive);
+    grid.change_state((1, 3), State::Alive);
+    grid.change_state((0, 2), State::Alive);
+
+    let mut last_updated = get_time();
+
+    let cube_width = 30.0;
+    let cube_height = 30.0;
+
+    let mut pause = true;
+
     loop {
-        clear_background(RED);
+        clear_background(GRAY);
+        if let Some(key) = get_last_key_pressed() {
+            keys[offset] = key;
+            offset += 1;
+            if offset == keys.len() {
+                offset = 0;
+            }
```

Comment:
What does this do?

Reply:
It's a secret ;)

(It's part of an easter egg 🐰 🥚 )